openedx/core/djangoapps/user_api/tests/test_models.py

```diff
@@ -83,3 +91,84 @@ def test_get_value(self):
         # get preference for key that doesn't exist for user
         pref = UserPreference.get_value(user, 'testkey_none')
         self.assertIsNone(pref)
+
+
+class TestUserPreferenceEvents(EventTestMixin, TestCase):
```

Comment:
Nit: could probably use UserSettingsEventTestMixin now instead of EventTestMixin, which would set the tracker name. It might also obviate the need for assert_user_preference_event_emitted.

Note that if so we have to define self.table in setup.